rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -2148,8 +2155,136 @@ def make_new_set():
         return set_dao.insert(new_set)
 
 
-def get_process_for_run(args, gcp_env):
+class UnblockSamples(ToolBase):
+    """
+    Unblocks a set of samples.
+    """
 
+    def __init__(self, args, gcp_env: GCPEnvConfigObject):
+        super().__init__(args, gcp_env)
+
+    def run(self):
+        self.gcp_env.activate_sql_proxy()
```

Comment:
In general, the readability of this method could be improved by breaking it into smaller chunks. I would split it into AW1, AW2, and Reconciliation methods and call them from the `run` method. There is cyclomatic complexity in this method that could be simplified. I've pointed out where I would simplify some of the if statements.